apple/RNGHVector.m

```diff
@@ -0,0 +1,67 @@
+//
+//  RNGHVector.m
+//  DoubleConversion
+//
+//  Created by Michał Bert on 05/08/2024.
+//
+
+#import "RNGHVector.h"
+#import <Foundation/Foundation.h>
+
+@implementation Vector
+
+- (id)initWithX:(double)x withY:(double)y
+{
+  if (self = [super init]) {
+    self.x = x;
+    self.y = y;
+
+    self.magnitude = hypot(x, y);
+
+    self.unitX = x / self.magnitude;
+    self.unitY = y / self.magnitude;
```

Comment:
I believe a division by `0` may happen here if a `x: 0, y: 0` `Vector` is constructed

Reply:
Maybe, but will we ever construct `[0,0]` vector? We have two ways to construct vectors, i.e. from `direction` and from `velocity`. Making vector from direction won't lead to **0**. In case of `velocity`, it is created inside `mouseDragged` and I think it won't be called if mouse stands in place (so that `dx = dy = 0`)

Reply:
On the other hand, there might be some calculations/hardware related issues so that it will be called and translation won't be registered. I'll fix that.

Reply:
> Making vector from direction won't lead to 0.
It may, that's the default case for an invalid value being passed.
Someone would have to explicitly pass an invalid value, but this may occur by accidentally i think, if someone was to combine conflicting directions like `Left | Right`, then i believe that would default to a `0, 0` vector.

In that case the error would be enigmatic and wouldn't help the user fix their issue.

>I think it won't be called if mouse stands in place (so that dx = dy = 0)

I wouldn't entirely count on that.
It's mostly my gut feeling though 😄

Reply:
> It may, that's the default case for an invalid value being passed.

The only possible values that we give to the users are in `RNGestureHandlerDirection` enum, so I don't think invalid value can be passed here.

>if someone was to combine conflicting directions like Left | Right, then i believe that would default to a 0, 0 vector.

Why would `Left | Right` be conflicting? It means that `Fling` should activate with swiping either `Left` or `Right`. What is the conflict here?

Reply:
>Why would Left | Right be conflicting? It means that Fling should activate with swiping either Left or Right. What is the conflict here?

`Left | Right` are conflicting only when directly passed to the `fromDirection`, but you're right that this will never occur because `Fling`'s implementations already prevents that from happening.

Here's how such situation would look like:
```
int conflicting = RNGestureHandlerDirectionRight | RNGestureHandlerDirectionLeft;
Vector *directionVector = [Vector fromDirection:conflicting];
```

The code above compiles with no warnings and results in creation of the following vector:
![image](https://github.com/user-attachments/assets/dd7203c1-cc89-4900-a9ab-0145a2a4d049)

Of course as you said, this situation is impossible with how the `Fling` handler works, and so it's not an immediate issue, but it's not preferable to have a global `Vector` class which could be used elsewhere, accepting invalid values without throwing any errors.

Reply:
Done in [6900228](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/6900228aa4663c8d2e327dcb5b8a5c39fbbeaaa5)